### DISCUSSION TITLE
Concerns about the futur of TOSCA implementations and solutions

### DISCUSSION THREAD

**Discussion by KaioKei** [tliron/go-puccini#84] [General]
Hello Mr Liron,

I have some questions regarding TOSCA and implementations of TOSCA orchestrator
frameworks. Especially I am not sure if the TOSCA approach is future proof or
if the momemtum arround TOSCA is over.

I do know solutions/implementations like Cloudify, Alien4Cloud + YORC,
Turandot+Puccini, TORCH, OpenStack/Tacker, Apache Brooklyn.
But most of these projects are proof of concepts and some of them are no longer
being maintained.

On the one hand, it seems TOSCA is "alive" : the OASIS standard is regularly
updated, but on the other hand, it is not easy to find documentation and recent content
and examples using TOSCA.

Questions:
1. What could encourage you to continue your work on your PoC Turandot+Puccini?
2. Is TOSCA a niche framework used only by telco operators?
3. Does TOSCA exist in the CNCF landscape https://landscape.cncf.io/?

**Comment by tliron** [comment]
I am a member of the TOSCA ad hoc committee where we are working hard on TOSCA 2.0. From our perspective there is a healthy momentum around TOSCA. I recommend watching our [TOSCA implementations series](https://www.youtube.com/watch?v=jusRsRrlKds&list=PLaYKtNo_BitYlGRp4NtuuPdd5zIrt9ZKV) to get a sense of the scope. You'll also see TOSCA used outside telco, for sure.

As for the CNCF landscape, it's tricky. I think Turandot could be a candidate for bringing TOSCA into Kubernetes, but it really needs to mature more before doing that.

How to get me to work more on these projects? Build a community around them! :) Have other projects that use these tools and invest in them. It's always harder to go alone.

**Reply by KaioKei** [reply]
Thank you very much for your answer ! :smiley: :+1: 
Thanks again for the link you have shared about TOSCA implementations series, I will certainly look into it.

Are you planning to upgrade Turandot to TOSCA 2.0 when its available or is it too soon to think about it ?

**Reply by tliron** [reply]
Absolutely! Since I am one of the key people working on TOSCA 2.0, and I'm very much about "code first", it's critical to me to implement 2.0 features in Puccini, and then to test them out in Turandot. From my perspective I want TOSCA 2.0 to embrace cloud native and that it needs to be proven. For my, ideas are useless if they stay theory, they must be proven in the real world.

By the way, OASIS is an open standards body and if you have an interest in being part of the standardization process you would be very welcome to join us.

**Reply by KaioKei** [reply]
This is wonderlful news !  :tada: 
I do agree with you about theories being weakened by the lack of proper implementations. 
This is why I am looking for well-established solutions with TOSCA standard for cloud-native orchestration and why I am so fustrated to find so few ... 

I really enjoy to hear from you that TOSCA 2.0 will probably embrace the CNCF perspective, so I can believe in the cloud native community to be more interested in the future in TOSCA too (play on words).                                                                                                                                                             
                                                                                                                                                                                    
Thank you for your prompt answers and I will probably consider your proposition to look more into the standardization process :slightly_smiling_face:

**Reply by tliron** [reply]
We maintain a [list of implementations here](https://github.com/oasis-open/tosca-community-contributions/wiki/Known-TOSCA-Implementations). Some of them are quite active!
